sound/soc/sof/intel/hda-codec.c

```diff
@@ -108,6 +134,12 @@ static int hda_codec_probe(struct snd_sof_dev *sdev, int address,
 
 	hda_priv->codec.bus = hbus;
 	hdev = &hda_priv->codec.core;
+	codec = &hda_priv->codec;
+
+	if (is_generic_config(hbus))
+		codec->probe_id = HDA_CODEC_ID_GENERIC;
+	else
+		codec->probe_id = 0;
```

Comment:
@kv2019i did you to say that the only way to use the generic model is through the use of a kernel module parameter?

I am not sure what prevents us from doing the same as legacy, and use the generic part as a fallback?

Reply:
@plbossart No, I just run out of time. I wanted to get a patch out to try by end-of-week and this was my late Friday effort for that. We should have proper fallback for generic and it's just a matter how to cleanly do that in SOF.

Reply:
ok, thanks for the precision!